copper-server-impl/src/main/java/com/pogeyan/cmis/impl/services/CmisVersioningServices.java

```diff
@@ -318,8 +318,15 @@ public static String checkIn(String repositoryId, Map<String, Object> properties
 					}
 				}
 			}
-			String fileName = documentObject.getContentStreamFileName()
-					+ documentObject.getVersionLabel().replace(".", "_");
+			String fileName;
+			if (documentObject.getContentStreamFileName().contains(".")) {
+				String[] fileNames = documentObject.getContentStreamFileName().split("\\.");
+				fileName = fileNames[0] + documentObject.getVersionLabel().replace(".", "_") + "." + fileNames[1];
```

Comment:
This will not work if the file has multiple . in the name, for ex: sample.text.one.txt (is a valid file name), Please use proper way to identify the name splits.